### ISSUE TITLE
Add support for npm aliases

### ISSUE THREAD

**Issue by watson** [elastic/apm-agent-nodejs#1270]
With npm cli v6.9.0, [npm now supports aliases](https://github.com/npm/rfcs/blob/latest/implemented/0001-package-aliases.md). This means that you can install more than one version of the same package. Users can then give each package their own name (the alias) which is then used as the argument to the `require` function instead of the actual name of the package.

The downside of this is that our require hook will not detect that the module is loaded because it purely looks at the string being passed into the `require` function.

Possible solutions:
- Allow the user to configure the agent with a mapping of aliases to actual module names.
- Implement a clever solution that parses the `package.json` file of the project to detect those aliases automatically.
- Read the `name` field of `package.json` of the module being required instead of relying on the name given to the `require` function.

**Comment by watson** [comment]
For a better example than given in the npm docs, see: https://github.com/elastic/elasticsearch-js/tree/update-events#install-multiple-versions

**Comment by Qard** [comment]
Hmm...seems complicated and potentially expensive--we'd have to hit the file system an extra time for _every_ require. Also, seems to me like this is more something that needs to happen in `require-in-the-middle`? Should the issue be moved there?

**Comment by delvedor** [comment]
If you install a package with an alias, for example:
```
npm install es6@npm:@elastic/elasticsearch@6
npm install es7@npm:@elastic/elasticsearch@7
```
This is how your `package.json` will look like:
```json
"dependencies": {
  "es6": "npm:@elastic/elasticsearch@^6.8.2",
  "es7": "npm:@elastic/elasticsearch@^7.3.0"
}
```
And this is how your `node_modules` will look like:
```sh
ll node_modules
drwxr-xr-x  11 delvedor  staff   352B July  2 09:28 es6
drwxr-xr-x  11 delvedor  staff   352B July  2 09:29 es7
```

As far as I know, the `require` algorithm search the folder name, am I right?
If so, you can provide a lookup table, and you should be able to handle the issue.
```js
require('elastic-apm-node').start({
  aliases: {
    es6: '@elastic/elasticsearch',
    es7: '@elastic/elasticsearch'
  }
})
```
And internally use this map to figure out which instrumentation to use.

**Comment by watson** [comment]
> Hmm...seems complicated and potentially expensive--we'd have to hit the file system an extra time for every require. Also, seems to me like this is more something that needs to happen in require-in-the-middle? Should the issue be moved there?

How about the config option suggestion? That doesn't require any I/O

**Comment by Qard** [comment]
How does it know what level to apply it at though? For example: the app, or a dependency, aliases `my-aliased-module` to `super-common-module-name` and then, somewhere else in the tree, the _actual_ `super-common-module-name` is loaded? How would it know _not_ to apply instrumentation to that one? Conversely, what if the alias matches something we have _our own_ instrumentation for? Do we then _not_ apply _our own_ instrumentation which matches the name due to the alias? Seems like this will get complicated very quickly...

**Comment by delvedor** [comment]
@Qard you are right. The `package.json` of the aliased module keeps the original package name, I think this might solve the issue.

**Comment by Qard** [comment]
Not saying it can't be done just that I don't think we can get around walking the `node_modules` folder and reading the `package.json` files.

**Comment by rh-danderson** [comment]
Hi! Just adding a scenario to help illustrate the challenge here:

We've got an API service that references multiple Elasticsearch clusters. When we need to upgrade their major version, we want to be able to upgrade one at a time, instead of having to coordinate a simultaneous upgrade of all of them.

Having an alias lets us address multiple major versions at the same time. It would be lovely if APM would continue to show us our queries.

For anyone facing this challenge, you can still get your queries to show for clusters of one of your cluster versions. In this example, we've chosen to get instrumentation for v7 clusters. 

`"elasticsearch-v6": "npm:@elastic/elasticsearch@^6.8.0",`
`"elasticsearch-v7": "npm:@elastic/elasticsearch@^7.6.1",`

and

`const { Client: ES6Client } = require("elasticsearch-v6");`
`const { Client: ES7Client } = require("elasticsearch-v7");`

becomes

`"elasticsearch-v6": "npm:@elastic/elasticsearch@^6.8.0",`
`"@elastic/elasticsearch": "7.6.1",`

and 

`const { Client: ES6Client } = require("elasticsearch-v6");`
`const { Client: ES7Client } = require("@elastic/elasticsearch");`
